src/huggingface_hub/hf_file_system.py

```diff
@@ -1069,6 +1069,7 @@ def read(self, length: int = -1):
             )
             hf_raise_for_status(self.response)
         try:
+            self.response.raw.decode_content = True
```

Comment:
can you also set this line on [L1095](https://github.com/huggingface/huggingface_hub/pull/3271/files#diff-9c58c269bb02cc610db29c7d3066a9a02cc7430fc4175fe917b9c53abac72d3dR1095) as well? we have a second read call where it would be beneficial